network.go

```diff
@@ -195,6 +195,12 @@ func (n *network) CreateEndpoint(name string, options ...EndpointOption) (Endpoi
 	if name == "" {
 		return nil, ErrInvalidName(name)
 	}
+
+	e, _ := n.EndpointByName(name)
```

Comment:
docker/docker committer had us change EndpointByName() to return an error in case the endpoint is not found.
Please don't discard the error and check it as it provides what you are looking for, instead of of checking for nil element.